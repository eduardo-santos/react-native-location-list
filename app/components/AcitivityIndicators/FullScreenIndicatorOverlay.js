import React from "react";
import PropTypes from "prop-types";
import { View, Text, ActivityIndicator } from "react-native";

import styles from "./styles";

const FullScreenIndicatorOverlay = ({ text }) => (
  <View style={styles.fullScreenOverlayContainer}>
    <ActivityIndicator size="large" color="#8fc0e8" />
    <Text style={styles.fullScreenOverlayText}>{text}</Text>
  </View>
);

FullScreenIndicatorOverlay.propTypes = {
  text: PropTypes.string
};

export default FullScreenIndicatorOverlay;
