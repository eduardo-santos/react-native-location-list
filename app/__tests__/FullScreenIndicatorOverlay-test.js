import React from "react";

import renderer from "react-test-renderer";
import FullScreenIndicatorOverlay from "../components/ActivityIndicators/FullScreenIndicatorOverlay";

it("renders correctly", () => {
  const tree = renderer.create(<FullScreenIndicatorOverlay />).toJSON();
  expect(tree).toMatchSnapshot();
});
