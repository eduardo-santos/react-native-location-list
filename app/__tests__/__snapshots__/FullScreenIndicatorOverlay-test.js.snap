// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.6)",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#8fc0e8"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "marginTop": 10,
      }
    }
  />
</View>
`;
